src/dotnet/APIView/APIViewWeb/Client/src/shared/signalr.ts

```diff
@@ -58,8 +53,270 @@ $(() => {
     hp.addCommentThreadNavigation();
     hp.removeCommentIconIfEmptyCommentBox(elementId);
+
+    let size = 28;
+    let $navLinks = $("nav.navbar a.nav-link");
+    let username;
+
+    for (let nav of $navLinks) {
+      if (nav.innerText.includes("Profile")) {
+        let href = nav.getAttribute("href");
+        if (href) {
+          let hrefString: string = href;
+          let hrefSplit = hrefString.split("/");
+          username = hrefSplit[hrefSplit.length - 1];
+        }
+      }
+    }
+
+    let url: string = "https://github.com/" + username + ".png?size=" + size;
+    $("div.review-thread-reply div.reply-cell img.comment-icon").attr("src", url);
   });
 
+  let approvalPendingText = "Current Revision Approval Pending";
+  let approvedByText = "Approved by:";
+  let approvesCurrentRevisionText = "Approves the current revision of the API";
+
+  connection.on("ReceiveApprovalSelf", (reviewId, revisionId, approvalToggle) => {
+    hp.disableButtonTemp("form.form-inline button.btn", 5000);
+
+    if (!checkReviewRevisionIdAgainstCurrent(reviewId, revisionId, true)) {
+      return;
+    }
+
+    let $approvalSpans: JQuery<HTMLElement> = $("#approveCollapse span.small.text-muted");
+
+    var indexResult = parseApprovalSpanIndex($approvalSpans, approvedByText, approvalPendingText, approvesCurrentRevisionText);
+    let upperTextSpan = indexResult["upperText"];
+
+    if (approvalToggle) {
+      removeUpperTextSpan(upperTextSpan, $approvalSpans);
+      addButtonApproval();
+    } else {
+      addUpperTextSpan(approvesCurrentRevisionText);
+      removeButtonApproval();
+    }
+  });
+
+  connection.on("ReceiveApproval", (reviewId, revisionId, approver, approvalToggle) => {
+    if (!checkReviewRevisionIdAgainstCurrent(reviewId, revisionId, true)) {
+      return;
+    }
+
+    let $approvalSpans: JQuery<HTMLElement> = $("#approveCollapse span.small.text-muted");
+
+    var indexResult = parseApprovalSpanIndex($approvalSpans, approvedByText, approvalPendingText, approvesCurrentRevisionText);
+    let approversIndex = indexResult["approvers"];
+
+    if (approversIndex === -1) {
+      return;
+    }
+
+    let lowerTextSpan: HTMLElement = $approvalSpans[approversIndex];
+    let approverHref = `/Assemblies/Profile/${approver}`;
+
+    if (approvalToggle) {
+      addApprover(lowerTextSpan, approvedByText, approverHref, approver);
+    } else {
+      removeApprover(lowerTextSpan, approver, approvalPendingText);
+    }
+  })
+
   // Start the connection.
   start();
 });
+
+/**
+ * Removes the @approver from @lowerTextSpan of review page
+ * @param lowerTextSpan HTMLElement of the span that contains who approved the review or pending approval
+ * @param approver GitHub username of the review approver
+ * @param approvalPendingText string of approval pending text to use when removing the last approver
+ */
+function removeApprover(lowerTextSpan: HTMLElement, approver: string, approvalPendingText: string) {
```

Comment:
I think you can move these functions to `review.module.ts` since they are operating on elements in the revieiw page.